ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/PerasCertDB/StateMachine.hs

```diff
@@ -97,7 +97,9 @@ instance StateModel Model where
         -- Do not add equivocating certificates.
         AddCert cert -> all p model.certs
          where
-          p cert' = perasCertRound cert /= perasCertRound cert' || cert == cert'
+          p cert' =
+            perasCertRound cert /= perasCertRound (validatedPerasCert cert')
+              || cert == validatedPerasCert cert'
```

Comment:
This predicate might be a bit contrived now, as we are comparing properties between both validated and unvalidated certs.

Maybe I'm being too paranoid, but this could potentially break some internal invariant if the cert validation becomes more complex in the future -- thinking concretely whether the notion of equality between `ValidatedPerasCert` will always imply equality between their inner `PerasCerts`.